AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/DefaultStorageTransferDatabase.swift

```diff
@@ -175,7 +175,9 @@ class DefaultStorageTransferDatabase {
 
             let pairs = self.linkTasksWithSessions(persistableTransferTasks: persistableTransferTasks, sessionTasks: sessionTasks)
             completionHandler(.success(pairs))
-            self.recoveryState = .completed
+            self.queue.async {
+                self.recoveryState = .completed
+            }
         }
 
         urlSession.getActiveTasks(resultHandler: sessionTaskHandler)
```

Comment:
Will this cause a deadlock?
 Line 155 checks if we are on `queue`, so line 183 is run on `queue`. And `URLsession.getActiveTasks(resultHandler: )` invokes again in the same queue?

Reply:
no, 183 runs on the queue, but the callback from urlSession.getActiveTasks is not on the queue, so putting lines 178 -180 on the queue is not recursive and allows the state to be safely updated.